File: TwoPointers/reversePrefix.js
/**
 * @param {string} word 
 * @param {string} ch 
 */
function reversePrefix(word, ch) {
    let right = word.indexOf(ch);
    
    if (right === -1) {
        return word;
    }

    let left = 0;
    const chars = word.split('');

    while (left < right) {
        [chars[left], chars[right]] = [chars[right], chars[left]];
        left++;
        right--;
    }

    return chars.join('');
}

console.log(reversePrefix("abcdefd", "d"))

File: TwoPointers/reverse_prefix.py
def reverse_prefix(word: str, ch: str) -> str:
    right = word.find(ch)

    if right == -1:
        return word
    
    left = 0
    chars = list(word)

    while left < right:
        chars[left], chars[right] = chars[right], chars[left]
        left += 1
        right -= 1

    return "".join(chars)



print(reverse_prefix("abcdefd", "d"))